### PR TITLE
Makes Button compatible with react router 4; closes #1274

### LINKED ISSUE #1274: Button component react router 4 compatibility
The Button component's path prop is designed for use with react router, but is currently incompatible with RR4

### Expected Behavior
When path prop is passed to button, a click should transition route

### Actual Behavior
When mounting component, javascript error (router.createPath is not a function) on this line:
https://github.com/grommet/grommet/blob/master/src/js/components/Button.js#L94

### Steps to Reproduce
 1. have react router > 4 installed
 2. pass path to button
3. try to render button, get error

### PR DESCRIPTION
**Pull request by zacksinclair**
Signed-off-by: Zack Sinclair <zacksinclair@gmail.com>

#### What does this PR do?
Makes Button compatible with React Router 4 w/ the same implementation as Anchor. 

#### What testing has been done on this PR?
All tests pass. Locally tested that this works for both RR3 and RR4.

#### How should this be manually tested?
Use Button with a path defined. Test that react router 3 / 4 both transition to the path

#### Any background context you want to provide?
@alansouzati made this nice and easy via his Anchor RR4 commit:
https://github.com/grommet/grommet/commit/74ee5fe625e7777d30d6c49098b72863a4dfd1a6

You'll notice this was mostly a cut and paste job.

#### What are the relevant issues?
#1274 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
Yes, in combination with 74ee5fe625e7777d30d6c49098b72863a4dfd1a6, release notes should state that grommet now supports react router 4 in both Anchor and Button components.

#### Is this change backwards compatible or is it a breaking change?
Yes backwards compatible

### CODE REVIEW

File: src/js/components/Button.js [hunk 1]
```diff
@@ -28,9 +28,9 @@ export default class Button extends Component {
     event.preventDefault();
 
     if ('push' === method) {
-      router.push(path);
+      (router.history || router).push(path.path || path);
```
Comment (alansouzati) [inline src/js/components/Button.js]: Button does not have the concept of `active`. So we don't need support for `path` as an object. it is just a regular string. So this line can be `(router.history || router).push(path)`

File: src/js/components/Button.js [hunk 2]
```diff
@@ -28,9 +28,9 @@ export default class Button extends Component {
...
     } else if ('replace' === method) {
-      router.replace(path);
+      (router.history || router).replace(path.path || path);
```
Comment (alansouzati) [inline src/js/components/Button.js]: same here

File: src/js/components/Button.js [hunk 3]
```diff
@@ -91,7 +91,17 @@ export default class Button extends Component {
       buttonLabel = <span className={`${CLASS_ROOT}__label`}>{label}</span>;
     }
 
-    let adjustedHref = (path && router) ? router.createPath(path) : href;
+    const target = path ? path.path || path : undefined;
```
Comment (alansouzati) [inline src/js/components/Button.js]: I believe we can get rid of target and use just path.

**Comment by alansouzati** [comment]
I added the changes @zacksinclair. Can you take an extra look?

**Comment by zacksinclair** [comment]
tested and confirmed your changes re. path as object; good catch and thanks for the quick merge

**Pull request merged**